file/src/main/scala/org/apache/pekko/stream/connectors/file/model.scala

```diff
@@ -25,11 +25,13 @@ object ArchiveMetadata {
   def create(filePath: String): ArchiveMetadata = new ArchiveMetadata(filePath)
 }
 
-final case class ZipArchiveMetadata(name: String) {
+final case class ZipArchiveMetadata(name: String, size: Long, lastModification: Instant) {
```

Comment:
this seems like it'd not be binary compatible, because of the generated methods for case classes that would change - I'm not sure if it's possible to manually 'add back' the original methods to restore binary compatibility?

Reply:
Ah, I missed that. Sure, I'll try to toss in some additional `apply` methods.

Reply:
Should we tag those `deprecated`, since we'd have to set `size` and `lastModification` to zero? I thought about that for the old `create()` method as well.

Reply:
I don't have a strong preference, so up to you as far as I'm concerned!